frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.tsx

```diff
@@ -92,23 +98,35 @@ class TagEditorParamInner extends Component<Props> {
     }
   }
 
-  setType = (type: TemplateTagType) => {
-    const {
-      tag,
-      parameter,
-      setTemplateTag,
-      setParameterValue,
-      setTemplateTagConfig,
-      originalQuestion,
-    } = this.props;
+  getTemplateTagConfig = (newType: TemplateTagType): ParameterValuesConfig => {
+    const { tag, parameter, originalQuestion } = this.props;
+    if (!parameter || !originalQuestion) {
+      return EMPTY_VALUES_CONFIG;
+    }
+
+    const query = originalQuestion.query();
+    const queryInfo = Lib.queryDisplayInfo(query);
+    if (!queryInfo.isNative) {
```

Comment:
This is the only real change. All functions below would throw for a non-native query.